framework/src/modules/chain/chain.js

```diff
@@ -217,16 +217,12 @@ module.exports = class Chain {
 					action.params[2]
 				),
 			getPeers: async action =>
-				this.scope.modules.peers.shared.getPeers(
-					action.params[0],
-					action.params[1]
-				),
+				promisify(this.scope.modules.peers.shared.getPeers)(action.params[0]),
```

Comment:
Maybe not relevant to this PR but as you tackled it already, the functions `generateDelegateList`, `getUnProcessedTransactions`, `getUnconfirmedTransactions` and `getMultisignatureTransactions` also need to be promisify'd.

Reply:
Ah good one, let's include this in the PR as we tackled half the problem here yet.